app/src/main/res/layout/fragment_invalid_password.xml

```diff
@@ -23,80 +23,100 @@
-            app:endIconDrawable="@drawable/ic_show_password"
-            app:endIconMode="password_toggle"
-            app:errorEnabled="true"
-            app:errorIconDrawable="@null"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/detachMailbox">
+            <TextView
+                android:id="@+id/enterPasswordDescription2"
+                style="@style/SettingHeader"
+                android:paddingTop="0dp"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/enterPasswordDescription1"
+                tools:text="@string/enterPasswordDescription2" />
+
+            <TextView
+                android:id="@+id/or"
+                style="@style/SettingHeader"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:padding="0dp"
+                android:paddingEnd="0dp"
+                android:text="@string/enterPasswordOrDescription"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/enterPasswordDescription2"
+                tools:ignore="RtlSymmetry" />
+
+            <com.google.android.material.button.MaterialButton
+                android:id="@+id/detachMailbox"
+                style="@style/TextButton"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:paddingHorizontal="@dimen/marginStandardVerySmall"
+                android:paddingVertical="0dp"
+                android:text="@string/buttonDetachMailbox"
+                app:layout_constraintBaseline_toBaselineOf="@id/or"
+                app:layout_constraintStart_toEndOf="@id/or" />
 
-            <com.google.android.material.textfield.TextInputEditText
-                android:id="@+id/passwordInput"
+            <com.google.android.material.textfield.TextInputLayout
+                android:id="@+id/passwordInputLayout"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:inputType="textPassword" />
+                android:layout_marginHorizontal="@dimen/marginStandardMedium"
+                android:hint="@string/enterPasswordTitle"
+                app:endIconContentDescription="@string/contentDescriptionButtonShowPassword"
+                app:endIconDrawable="@drawable/ic_show_password"
+                app:endIconMode="password_toggle"
+                app:errorEnabled="true"
+                app:errorIconDrawable="@null"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/detachMailbox">
```

Comment:
We should add a top margin before this text input field (to match the specs but also to avoid missclicks on the "detach address" button). Make it so the field is constrained to the bottom of the 3rd paragraph instead of the end of the button which has a size which is less "under control"